src/foam/nanos/crunch/ui/CapabilityView.js

```diff
@@ -0,0 +1,105 @@
+foam.CLASS({
+  package: 'foam.nanos.crunch.ui',
+  name: 'MyView',
+  extends: 'foam.u2.View',
+  documentation: 'All purpose capability view that takes in an array of capaiblity ids and display these capabilities',
+
+  imports: [
+    'crunchController',
+    'notify'
+  ],
+
+  requires: [
+    'foam.u2.detail.SectionView'
+  ],
+
+  properties: [
+    {
+      class: 'StringArray',
+      name: 'capabilities'
+    },
+    {
+      name: 'wizardlets',
+      documentation: 'wizardlets for the capability'
+    },
+    {
+      name: 'wizardletSectionsList',
+      documentation: `
+        sections for wizardlets
+        wizardletSectionsList[i] stores sections for wizardlets[i]
+      `
+    }
+  ],
+
+  methods: [
+    async function initE() {
+      const self = this;
+
+      this.start().addClass(this.myClass())
+        .add(this.slot(function(capabilities) {
+          return this.E().forEach(capabilities, async function(cap) {
+            // get wizardlets for the capability and all of its dependent capabilities
+            self.wizardlets = (await self.crunchController.getCapsAndWizardlets(cap)).wizCaps;
+
+            // add listeners on wizardlets
+            for ( const wizardlet of self.wizardlets ) {
```

Comment:
sorry for being a burden, but const is an interesting choice. It works, however const creates a new block scope with each iteration ... 'let' would be my choice.